src/pkg/factory.go

```diff
@@ -19,7 +19,19 @@ import (
 
 	"github.com/goharbor/harbor/src/lib/config"
 	"github.com/goharbor/harbor/src/pkg/artifact"
-	"github.com/goharbor/harbor/src/pkg/cached/artifact/redis"
+	cachedArtifact "github.com/goharbor/harbor/src/pkg/cached/artifact/redis"
+	cachedProject "github.com/goharbor/harbor/src/pkg/cached/project/redis"
+	"github.com/goharbor/harbor/src/pkg/project"
+)
+
+type initManagerFunc func(cacheEnable bool)
+
+var (
+	// initManagerFuncs is the registry for init manager func.
+	initManagerFuncs = []initManagerFunc{
```

Comment:
Not clear about the intention of initManagerFuncs.  A slice will make us consider it is configurable,  for example if I want to disable cache in ArtifactManager, just removing the initArtifactManager from slice, but the current code doesn't work in this way.
Suggest to remove this initManagerFuncs and call it in the init function of current package
```
func init(){
   initArtifactManager(true)
   initProjectManager(true)
}
```

Reply:
updated